### PR TITLE
Add readthedocs/mkdocs integration to the documentation

### LINKED ISSUE #115: Add readthedocs/mkdocs integration to the documentation
We can use the work done on https://github.com/argoproj-labs/argocd-image-updater/ as a template:
- Create a readthedocs account, with a name like `argo-cd-application-set`
- Add `.readthedocs.yml`, `mkdocs.yml`, and additional metadata from `/docs` like `requirements.txt`, plus `css`, `js`
- Import that into read the docs and iterate on it until it looks good :)

I have a prototype branch which seems to be working, so this should be straightforward.

### PR DESCRIPTION
**Pull request by jgwest**
You can see the read the docs integration live here: https://argocd-applicationset.readthedocs.io/
Fixes #115 

## PR Description

Add readthedocs integration files based on argocd-image-updater
Add Docker-based mkdocs integration/linting from Argo CD

### Update repo README:
- Add links to new read the docs page
- Move Development section into docs
- De-emphasize design proposal link

### Documentation itself:
- Fixed differences between GitHub-flavoured markdown and mkdocs-flavoured markdown
- Add admonitions (lines beginning with `!!! note` or `!!! warning`)
- Read through and tweaked awkward sentences/phrasing (I still professionals to take a look at this :) )
- Moved Development section into docs
- Fixed mkdocs linter errors

### Known issues:
- For unknown reasons, the YAML syntax highlighting isn't enabled on YAML code blocks. I'll file an issue for this and handle it in a separate PR.

### CODE REVIEW

File: docs/Argo-CD-Integration.md [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 
-Argo CD generates the application's Kubernetes resources based on to the contents of the Git repository defined within the Application `spec`, deploying e.g. Deployments, Service, and other resources.
+Argo CD generates the application's Kubernetes resources based on the contents of the Git repository defined within the Application `spec` field, deploying e.g. Deployments, Service, and other resources.
 
-Creation, updates, or deletions of ApplicationSets will have a direct effect on the Applications present in the Argo CD namespace. Likewise, cluster events (the addition/deletion of Argo CD cluster secrets, when using Cluster generator), or changes in Git (when using Git generator), will be used as input to the ApplicationSet controller in constructing `Application` resources.
+Creation, updates, or deletion of ApplicationSets will have a direct effect on the Applications present in the Argo CD namespace. Likewise, cluster events (the addition/deletion of Argo CD cluster secrets, when using Cluster generator), or changes in Git (when using Git generator), will be used as input to the ApplicationSet controller in constructing `Application` resources.
```
Comment (wtam2018) [inline docs/Argo-CD-Integration.md:23]: Should "updates" be "update" (singular)?

File: docs/Template.md [hunk 2]
```diff
@@ -23,18 +23,21 @@ A template from a Cluster generator:
...
-  - **server**: API Server URL for the cluster (Example: `https://kubernetes.default.svc`)
-  - **namespace**: Target namespace in which to deploy the manifests from `source` (Example: `my-app-namespace`)
-  - Note:
-    - Referenced clusters must already be defined in Argo CD, for ApplicationSet controller to use them
-    - Only **one** of `name` or `server` may be specified: if both are specified, an error is returned.
+    - **name**: Name of the cluster (within Argo CD) to deploy to
+    - **server**: API Server URL for the cluster (Example: `https://kubernetes.default.svc`)
+    - **namespace**: Target namespace in which to deploy the manifests from `source` (Example: `my-app-namespace`)
+
+But note:
```
Comment (wtam2018) [inline docs/Template.md:37]: "But notes" sounds a bit strange to me.   Drop the "But"?

File: docs/Argo-CD-Integration.md (hunk 1 above)
Comment (jgwest) [reply docs/Argo-CD-Integration.md]: Done.

File: docs/Template.md (hunk 2 above)
Comment (jgwest) [reply docs/Template.md]: Done.

**Review by wtam2018 — APPROVED** [review]
LGTM
@jgwest Thanks.

**Pull request merged**